lib/ansible/modules/cloud/ovirt/ovirt_vm.py

```diff
@@ -1265,8 +1270,9 @@ def __get_template_with_version(self):
         template = None
         templates_service = self._connection.system_service().templates_service()
         if self.param('template'):
+            cluster = self.param('template_cluster') if self.param('template_cluster') else self.param('cluster')
             templates = templates_service.list(
-                search='name=%s and cluster=%s' % (self.param('template'), self.param('cluster'))
+                search='name=%s and cluster=%s' % (self.param('template'), cluster)
```

Comment:
If user don't specify `template_cluster`, and template is not found in `cluster`, we should fallback to global search